### PR TITLE
allowing syntax as One(as(1.0f)) to use a constant

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: test/constant/scalar/one.cpp [hunk 1]
```diff
@@ -23,4 +23,5 @@ STF_CASE_TPL( "Check one behavior"
   using boost::simd::One;
 
   STF_EQUAL(One<T>(), T(1));
+  STF_EQUAL(One(as(T(1))), T(1));
```
Comment (jfalcou) [inline test/constant/scalar/one.cpp]: could have been less whimsical if the example here was `STF_EQUAL(One(as(T(42))), T(1));`
So we don't think the 1 is propagated

Comment (jtlap) [reply test/constant/scalar/one.cpp]: Ok

Comment (jfalcou) [reply test/constant/scalar/one.cpp]: I also think a point in the doc about the new syntax is worthwhile. 
Do we have a common Constant.hpp pages in the doc ?

File: include/boost/simd/constant.hpp [hunk 2]
```diff
@@ -21,8 +21,16 @@ namespace boost { namespace simd
...
-      so that the correct type of the required constant is returned.
+      Constant generator functions ced in two manners:
+
+        - taking no arguments but requiring a template type parameter
+        so that the correct type of the required constant is returned
+        (as @c One<float>() to generate 1.0f).
+
+        - taking one dummy arguments from which  the correct type of the
+        required constant is deduced and  returned (as @c One(42.0f) to generate 1.0f).
+
```
Comment (jfalcou) [inline include/boost/simd/constant.hpp:32]: Maybe put an actual example ike in the other block ?

Comment (jtlap) [reply include/boost/simd/constant.hpp:32]: yes I will do it later in a doc related PR, with the example in the right directory

**Pull request merged**